artifact/sha/platform/openssl/sha.cpp

```diff
@@ -140,5 +142,22 @@ ExpectedSHA Reader::ShaSum() {
 	return SHA(hash, SHA_256_digest_length);
 }
 
+ExpectedSHA Shasum(const vector<uint8_t> &data) {
+	string in {data.begin(), data.end()};
+
+	io::StringReader is {in};
+
+	Reader r {is, ""};
```

Comment:
Observation, I don't think you need to pass in a string to the constructor at all, if you don't expect a shasum check

Reply:
Right. The problem here is that from this compile unit `Reader r {}` is ambiguous because both declared constructors `Reader(io::Reader &reader)` and `Reader(io::Reader &reader, const std::string &expected_sha = "")` are invoked the same way. This was not a problem when this function was part of `crypto` library, from the outside there is no ambiguity (`.hpp` declaration is with no default value).

I'll remove the default value in the constructor :+1: